tasks/install.yml

```diff
@@ -0,0 +1,80 @@
+---
+
+- name: Ensure Jenkins dependencies are installed
+  become: yes
+  package:
+    name: "{{ item }}"
+    state: installed
+  with_items:
+    - apt-transport-https
+    - curl
+
+- name: Ensure pre-requisite Jenkins directories
+  become: yes
+  file:
+    group: "{{ jenkins_server.group }}"
+    owner: "{{ jenkins_server.user }}"
+    path: "{{ item.value }}"
+    state: directory
+  with_dict: "{{ jenkins_server.directories }}"
+
+- name: Download Jenkins package
+  become: yes
+  get_url:
+    dest: "/usr/local/src/jenkins_{{ jenkins_server.version }}_all.deb"
+    sha256sum: "{{ jenkins_server.download_checksum }}"
+    url: "{{ jenkins_server.download_base_url }}/jenkins_{{ jenkins_server.version }}_all.deb"
+
+- name: Install Jenkins
+  become: yes
+  apt:
+    deb: "/usr/local/src/jenkins_{{ jenkins_server.version }}_all.deb"
+    state: installed
+  register: jenkins_service_installation
+
+- name: Setup Jenkins service defaults
+  become: yes
+  template:
+    dest: /etc/default/jenkins
+    src: service.default.j2
+  register: jenkins_service_defaults
+
+- name: Ensure Jenkins is restarted if defaults updated
+  become: yes
+  service:
+    name: jenkins
+    state: restarted
+  when: jenkins_service_defaults.changed
+
+- name: Wait for Jenkins to start up
+  include: wait_for_restart.yml
+  when: jenkins_service_defaults.changed
+
+- name: Check if downloaded Jenkins CLI from local server
+  become: yes
+  become_user: "{{ jenkins_server.user }}"
+  stat:
+    path: "{{ jenkins_server.directories.home }}/jenkins-cli.jar"
+  register: jenkins_service_cli
+
+- name: Download Jenkins CLI from local server
+  become: yes
+  become_user: "{{ jenkins_server.user }}"
+  get_url:
+    dest: "{{ jenkins_server.directories.home }}"
+    url: "http://{{ jenkins_server.hostname_local }}:{{ jenkins_server.http_port }}/jnlpJars/jenkins-cli.jar"
+  when: not jenkins_service_cli.stat.exists
+
+# Use CLI as Ansible module seems to have issues with newer Jenkins versions
+- name: Install Jenkins plugins via CLI
```

Comment:
Ansible does it with jenkins_plugin

Reply:
Doesn't work properly with newer Jenkins versions (as mentioned in the comment :) )

Reply:
jenkins_server:
  plugins:
    ansicolor:
      version: "0.4.3"
      enabled: yes